lntemp/harness_assertion.go

```diff
@@ -952,16 +956,14 @@ func (h *HarnessTest) ReceivePaymentUpdate(
 	select {
 	case <-time.After(DefaultTimeout):
 		require.Fail(h, "timeout", "timeout waiting for payment update")
+		return nil, nil
 
 	case err := <-errChan:
-		require.Failf(h, "payment stream",
-			"received err from payment stream: %v", err)
+		return nil, err
```

Comment:
I think it's because we want to put the method inside a `wait` to retry if an error is returned. Also we might want to test the error returned from the stream.